tests/gpt_j.rs

```diff
@@ -0,0 +1,89 @@
+use rust_bert::gpt_j::{
+    GptJConfigResources, GptJMergesResources, GptJModelResources, GptJVocabResources,
+};
+use rust_bert::pipelines::common::ModelType;
+use rust_bert::pipelines::text_generation::{TextGenerationConfig, TextGenerationModel};
+use rust_bert::resources::RemoteResource;
+use tch::Device;
+
+/// Equivalent Python code:
+///
+/// ```python
+/// import torch
+/// from transformers import AutoTokenizer, GPTJForCausalLM
+///
+/// device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+///
+/// model = GPTJForCausalLM.from_pretrained("anton-l/gpt-j-tiny-random").to(device)
+///
+/// tokenizer = AutoTokenizer.from_pretrained("anton-l/gpt-j-tiny-random", padding_side="left")
+/// tokenizer.add_special_tokens({"pad_token": tokenizer.eos_token})
+///
+/// inputs = [
+///     "It was a very nice and sunny",
+///     "It was a gloom winter night, and"
+/// ]
+///
+/// prompts = tokenizer(inputs, return_tensors="pt", padding=True, truncation=True).to(device)
+///
+/// with torch.no_grad():
+///     gen_tokens = model.generate(
+///         **prompts,
+///         min_length=0,
+///         max_length=32,
+///         do_sample=False,
+///         early_stopping=True,
+///         num_beams=1,
+///         num_return_sequences=1
+///     )
+///
+/// gen_texts = tokenizer.batch_decode(gen_tokens, skip_special_tokens=True)
+/// ````
+#[test]
+fn test_generation_gpt_j() -> anyhow::Result<()> {
```

Comment:
Thank you for adding tests. 

I would suggest the following:
- For the `tiny` test that will run in the pipeline, please instead do a forward pass through the model by passing an input tensor and a past tensor, and compare that to the output generated by Python (e.g. to the 4th decimal, see https://github.com/guillaume-be/rust-bert/blob/78da0f4814da00139551b615d6d06e7e92f59a79/tests/m2m100.rs#L64 for example)
- It would be nice to add a test for a full-sized model. Can you add a test that generates a sensible output to test the generation process using float16, on CPU and flagging the test as `#[cfg_attr(not(feature = "all-tests"), ignore)]` )?

Reply:
I'll work on the test you suggested for the `tiny` model.

As for the full-sized model using float16 it won't run on the CPU, when I tried I got a Torch error: "LayerNormKernelImpl" not implemented for 'Half'. I think some cuda kernels are not available for inference on the CPU when using float16. Using the folat32 version requires at least 32 Gb of RAM and I'm not sure the CI has that much available. What do you think ?